hasura.planx.uk/migrations/1702916788570_alter_column_analytics_logs_node_type/up.sql

```diff
@@ -0,0 +1,70 @@
+-- ALTER COLUMN will fail if dependent view exists
+DROP VIEW public.analytics_summary;
+
+-- Update column type from integer to text
+ALTER TABLE public.analytics_logs
+ALTER COLUMN node_type TYPE TEXT 
+USING node_type::text;
+
+-- Do a one-time update of historic records
+UPDATE public.analytics_logs SET node_type = 
+    CASE 
+        WHEN node_type = '1' THEN 'Flow'
+        WHEN node_type = '3' THEN 'Result'
+        WHEN node_type = '7' THEN 'TaskList'
+        WHEN node_type = '8' THEN 'Notice'
+        WHEN node_type = '9' THEN 'FindProperty'
+        WHEN node_type = '10' THEN 'DrawBoundary'
+        WHEN node_type = '11' THEN 'PlanningConstraints'
+        WHEN node_type = '12' THEN 'PropertyInformation'
+        WHEN node_type = '100' THEN 'Statement'
```

Comment:
This has always felt like a bit of a tricky one - if we get feedback that this is oddly named we can very easily update to "Question" or something in future 👍

Reply:
Yep agree - that's probably a change we should consider making at the root type definition rather than just analytics! Hopefully ambiguous text is still a huge upgrade from integer for starters though!